gprofiler/profilers/perf.py

```diff
@@ -44,7 +44,7 @@
 # ffffffff81082227 mmput+0x57 ([kernel.kallsyms])
 # 0 [unknown] ([unknown])
 # 7fe48f00faff __poll+0x4f (/lib/x86_64-linux-gnu/libc-2.31.so)
-FRAME_REGEX = re.compile(r"^\s*[0-9a-f]+ (.*?) \(\[?(.*?)\]?\)$")
+FRAME_REGEX = re.compile(r"^\s*[0-9a-f]+ (.*) \(\[?(.*?)\]?\)$")
```

Comment:
The regex needs to get more complex.
We need hints to avoid matching parenthesis in dso name while allowing `(deleted)` output in library path resolution.
I propose this, as a starting point:
```suggestion
FRAME_REGEX = re.compile(r"^\s*[0-9a-f]+ (?P<symbol>.*?) \(\[?(?P<library_name>[^(]*?(?:\(deleted\))?)\]?\)$")
```

Reply:
Hmm, might work. Can you check how this affects performance? e.g take a few example stacks from `perf scripts` and use `timeit` to see how much time `_collapse_stack` takes with the changed regex. It is a very critical part in the hot path so I want to make sure we are not worsening performance.

Reply:
That's 2us vs 1.4us for improved vs the old regex.
Sample file: 520k lines from perf script.